src/crypto/XDRHasher.h

```diff
@@ -35,8 +35,11 @@ template <typename Derived> struct XDRHasher
                 return;
             }
         }
-        memcpy(mBuf + mLen, u, sz);
-        mLen += sz;
+        if (sz != 0)
+        {
+            memcpy(mBuf + mLen, u, sz);
```

Comment:
My reading of the spec is that UB is when the pointers are invalid/null, `sz == 0` by itself is not UB. As `mBuf` is a valid pointer, we should just have a guard at the top of that function against `u == nullptr`

Reply:
> My reading of the spec is that UB is when the pointers are invalid/null, `sz == 0` by itself is not UB. As `mBuf` is a valid pointer, we should just have a guard at the top of that function against `u == nullptr`

That's the thing: u is sometimes nullptr (exactly when sz == 0). I can _branch_ on either, but I can't exclude either as a possibility (unless it's a bug in xdrpp -- looking).

Reply:
Yeah this isn't a bug in xdrpp, it's just a consequence of the way we're called when processing a string field that's optional / missing (or otherwise zero-length). I can branch on nullptr and assert zero size, if you prefer that?